tests/performance_tests/test_memory_usage.py

```diff
@@ -39,8 +39,8 @@ def poly_template(monkeypatch):
     yield folder
 
 
-@pytest.mark.flaky(reruns=5)
-@pytest.mark.limit_memory("130 MB")
+# @pytest.mark.flaky(reruns=5)
+@pytest.mark.limit_memory("1 GB")
```

Comment:
Is there any way to lower the requirement to make this test more targeted? It is intended to test the memory use of the update step, but now it is mostly testing how much memory creating the S matrix takes.

Reply:
I think it might be a bit difficult as it calls into `smoother_update->analysis_ES->_load_observations_and_responses`, which is where the S-matrix is constructed. If there is some way to tell mark.limit_memory to ignore the spike arising from `ensemble.get_measured_data([*selected_observations], iens_active_index)` that would be the absolute ideal I suppose?

Reply:
Some profiling:
```
--combine--
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    46    293.8 MiB    293.8 MiB           1   @profile
    47                                         def test_memory_smoothing(poly_template):
    48    659.3 MiB    365.5 MiB           1       ert_config = ErtConfig.from_file("poly.ert")
    49   1135.2 MiB    475.9 MiB           1       fill_storage_with_data(poly_template, ert_config)
    50   1362.2 MiB   -162.2 MiB           2       with open_storage(poly_template / "ensembles", mode="w") as storage:
    51   1135.3 MiB      0.0 MiB           1           prior_ens = storage.get_ensemble_by_name("prior")
    52   1137.9 MiB      2.7 MiB           2           posterior_ens = storage.create_ensemble(
    53   1135.3 MiB      0.0 MiB           1               prior_ens.experiment_id,
    54   1135.3 MiB      0.0 MiB           1               ensemble_size=prior_ens.ensemble_size,
    55   1135.3 MiB      0.0 MiB           1               iteration=1,
    56   1135.3 MiB      0.0 MiB           1               name="posterior",
    57   1135.3 MiB      0.0 MiB           1               prior_ensemble=prior_ens,
    58                                                 )
    59   1524.4 MiB    386.5 MiB           2           smoother_update(
    60   1137.9 MiB      0.0 MiB           1               prior_ens,
    61   1137.9 MiB      0.0 MiB           1               posterior_ens,
    62   1137.9 MiB      0.0 MiB           1               str(uuid.uuid4()),
    63   1137.9 MiB      0.0 MiB           1               list(ert_config.observation_keys),
    64   1137.9 MiB      0.0 MiB           1               list(ert_config.ensemble_config.parameters),
    65                                                 )



--main--
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    46    300.1 MiB    300.1 MiB           1   @profile
    47                                         def test_memory_smoothing(poly_template):
    48    361.5 MiB     61.4 MiB           1       ert_config = ErtConfig.from_file("poly.ert")
    49    372.5 MiB     11.0 MiB           1       fill_storage_with_data(poly_template, ert_config)
    50    421.9 MiB     -9.9 MiB           2       with open_storage(poly_template / "ensembles", mode="w") as storage:
    51    373.6 MiB      0.0 MiB           1           prior_ens = storage.get_ensemble_by_name("prior")
    52    373.8 MiB      0.1 MiB           2           posterior_ens = storage.create_ensemble(
    53    373.6 MiB      0.0 MiB           1               prior_ens.experiment_id,
    54    373.6 MiB      0.0 MiB           1               ensemble_size=prior_ens.ensemble_size,
    55    373.6 MiB      0.0 MiB           1               iteration=1,
    56    373.6 MiB      0.0 MiB           1               name="posterior",
    57    373.6 MiB      0.0 MiB           1               prior_ensemble=prior_ens,
    58                                                 )
    59    432.9 MiB     59.1 MiB           2           smoother_update(
    60    373.8 MiB      0.0 MiB           1               prior_ens,
    61    373.8 MiB      0.0 MiB           1               posterior_ens,
    62    373.8 MiB      0.0 MiB           1               str(uuid.uuid4()),
    63    373.8 MiB      0.0 MiB           1               list(ert_config.observations.keys()),
    64    373.8 MiB      0.0 MiB           1               list(ert_config.ensemble_config.parameters),
    65                                                 )
```

Reply:
I'm not sure if it can be done. The update step does create the S matrix. The table join operation in doing so, between the responses and observations on the same primary key, does consume more memory than opening up one response ds at a time. It would mitigate this if we added chunking to the datasets, but that introduces a bit more complexity also, would also have to test on NFS. Running it on my machine with some chunking on the responses dataset, `{"name": 1000}` is significantly slower, takes 28s with chunking and in total it allocates a bit more memory (not sure on the specifics of how it can be done), as opposed to 18s without the chunking. I think (don't know for 100% sure though) that turning them into dask-dataarrays under the hood (which happens when using chunks) will introduce quite some overhead.

Reply:
How much memory does ErtConfig hold in Drogon on main vs. combine?

Reply:
Tried on `drogon_ahm`, seems not so significant, only significant if there are many observations
```Combined
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    80    328.0 MiB    328.0 MiB           1   @profile
    81                                         def _start_initial_gui_window(
    82                                             args, log_handler, plugin_manager: Optional[ErtPluginManager] = None
    83                                         ):
...
   100    345.9 MiB     17.9 MiB           1               ert_config = ErtConfig.from_file(args.config)
...

Main

Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
    80    325.7 MiB    325.7 MiB           1   @profile
    81                                         def _start_initial_gui_window(
    82                                             args, log_handler, plugin_manager: Optional[ErtPluginManager] = None
    83                                         ):
...
   100    340.2 MiB     14.3 MiB           1               ert_config = ErtConfig.from_file(args.config)
...```